File: CPP/2903FindIndicesWithIndexAndValueDifferenceI.cpp
#include <iostream>
#include <vector>
#include  <cmath>

using namespace std;

vector<int> findIndices(vector<int>& nums, int indexDifference, int valueDifference){
    int n = nums.size();
    vector<int> defaultResult = {-1, -1};

    for(int i = 0; i < n; i++){
        for(int j = 0; j < n; j++){
            if(abs(i - j) >= indexDifference){
                if(abs(nums[i] - nums[j]) >= valueDifference){
                    vector<int> result = {i, j};
                    return result;
                }
            }
        }
    }

    return defaultResult;
}

int main() {

    vector<int> nums = {5, 1, 4, 1};
    int indexDifference = 2;
    int valueDifference = 4;

    vector<int> result = findIndices(nums, indexDifference, valueDifference);
    cout << "[" << result[0] << ", " << result[1] << "]" << endl;
    return 0;
}

File: Python/2903FindIndicesWithIndexAndValueDifferenceI.py
def findIndices(nums : list[int], indexDifference : int, valueDifference : int) -> list[int]:
    n = len(nums)

    for i in range(n):
        for j in range(i + 1, n):
            if ((abs(i - j)) >= indexDifference):
                if ((abs(nums[i] - nums[j]) >= valueDifference)):
                    return [i, j]
    
    return [-1, -1]


nums = [5, 1, 4, 1] 
indexDifference = 2 
valueDifference = 4

print(findIndices(nums, indexDifference, valueDifference))